epsagon/utils.py

```diff
@@ -80,12 +91,39 @@ def init(token,
     :return: None
     """
     if not collector_url:
-        collector_url = get_tc_url(use_ssl)
+        collector_url = get_tc_url(
+            (get_env_or_val('EPSAGON_SSL', '') == 'TRUE') | use_ssl
+        )
     tracer.initialize(
-        token=token,
-        app_name=app_name,
-        collector_url=collector_url,
-        metadata_only=metadata_only,
-        use_ssl=use_ssl,
-        debug=debug
+        token=get_env_or_val('EPSAGON_TOKEN', token),
+        app_name=get_env_or_val('EPSAGON_APP_NAME', app_name),
+        collector_url=get_env_or_val('EPSAGON_COLLECTOR_URL', collector_url),
+        metadata_only=(
+          (get_env_or_val('EPSAGON_METADATA', '') == 'TRUE') | metadata_only
+        ),
+        debug=(get_env_or_val('EPSAGON_DEBUG', '') == 'TRUE') | debug
     )
+
+
+def import_original_module():
```

Comment:
look at bootstrap.py

Reply:
```
def _get_handlers(handler, mode):
    lambda_runtime.report_user_init_start()
    init_handler = lambda: None

    """
    This is the old way we were loading modules.
    It was causing intermittent build failures for unknown reasons.
    Using the imp module seems to remove these failures.
    The imp module appears to be more extreme in that it reloads
    the module if it is already loaded, and it likely doesn't use any caches when
    searching for the module but does a full directory search, which is what we want.
    """
    # m = imp.load_module(modname, globals(), locals(), [])

    try:
        (modname, fname) = handler.rsplit('.', 1)
    except ValueError as e:
        fault = wsgi.FaultException("Bad handler '{}'".format(handler), str(e), None)
        request_handler = make_fault_handler(fault)
        lambda_runtime.report_user_init_end()
        return init_handler, request_handler

    file_handle, pathname, desc = None, None, None
    try:
        # Recursively loading handler in nested directories
        for segment in modname.split('.'):
            if pathname is not None:
                pathname = [pathname]
            file_handle, pathname, desc = imp.find_module(segment, pathname)
        if file_handle is None:
            module_type = desc[2]
            if module_type == imp.C_BUILTIN:
                request_handler = make_fault_handler(wsgi.FaultException(
                    "Cannot use built-in module {} as a handler module".format(modname),
                    None,
                    None
                ))
                lambda_runtime.report_user_init_end()
                return init_handler, request_handler
        m = imp.load_module(modname, file_handle, pathname, desc)
    except Exception as e:
        request_handler = load_handler_failed_handler(e, modname)
        lambda_runtime.report_user_init_end()
        return init_handler, request_handler
    finally:
        if file_handle is not None:
            file_handle.close()

    try:
        init_handler = getattr(m, 'init')
    except AttributeError as e:
        pass

    try:
        request_handler = make_final_handler(getattr(m, fname), mode)
    except AttributeError as e:
        fault = wsgi.FaultException("Handler '{}' missing on module '{}'".format(fname, modname), str(e), None)
        request_handler = make_fault_handler(fault)
    lambda_runtime.report_user_init_end()
    return init_handler, request_handler
```

Reply:
Their code is not really nice, and it uses imp that uses importlib, that uses __import__